stor/s3.py

```diff
@@ -765,3 +769,34 @@ def to_url(self):
         """Returns HTTP url for object (virtual host-style)"""
         return u'https://{bucket}.s3.amazonaws.com/{key}'.format(bucket=self.bucket,
                                                                  key=self.resource)
+
+    def restore(self, tier='Bulk', days=10):
+        """Issue a restore command for a single object from glacier.
+
+        Args:
+            tier (str, optional): restore speed (see Glacier docs for details)
+            days (int, optional): number of days to keep data in S3 post-restore.
+        Returns:
+            dict or None: restore response from S3 client
+        Note:
+            Calling ``restore()`` on a directory will not work correctly. Only use this for single
+            objects! (you can always do ``[s.restore() for s in stor.list(<directory>)]``)
+
+        Ignores RestoreAlreadyInProgressError and AlreadyRestoredError (note that you can't force
+        S3 to do a faster restore once you've chosen a tier)
+        """
+        valid_tiers = ('Standard', 'Bulk', 'Expedited')
+        if tier not in valid_tiers:
+            raise TypeError('`tier` must be one of {}'.format(valid_tiers))
```

Comment:
TypeError is not the best exception type in this case: https://docs.python.org/2/library/exceptions.html#exceptions.TypeError

ValueError would be better.